app/ui/src/main/java/de/mm20/launcher2/ui/launcher/search/location/LocationItem.kt

```diff
@@ -0,0 +1,306 @@
+                                            }
+                                    } ?: location.openingHours!!.first()
+
+                            val openIndex = location.openingHours!!.indexOfFirst { it.isOpen }
+                            val timeFormatter = DateTimeFormatter.ofLocalizedTime(FormatStyle.SHORT)
+                            if (openIndex != -1) {
+                                Text(
+                                    text = stringResource(
+                                        R.string.location_open_until,
+                                        (LocalTime.now() + location.openingHours!![openIndex].duration).format(
+                                            timeFormatter
+                                        )
+                                    )
+                                )
+                            }
+                            Text(
+                                text = stringResource(
+                                    if (nextOpeningTime.dayOfWeek == today) R.string.location_open_next
+                                    else R.string.location_open_next_day,
+                                    if (nextOpeningTime.dayOfWeek == today) Duration.between(
+                                        nextOpeningTime.startTime,
+                                        LocalTime.now()
+                                    )
+                                    else "${
+                                        nextOpeningTime.dayOfWeek.getDisplayName(
+                                            TextStyle.SHORT,
+                                            Locale.getDefault()
+                                        )
+                                    } ${nextOpeningTime.startTime.format(timeFormatter)}"
+                                )
+                            )
+                        }
+
+                        val userHeading by viewModel.trueNorthHeading.collectAsState(null)
+                        if (userLocation != null && userHeading != null) {
+                            val directionArrowAngle by animateFloatAsState(
+                                targetValue = userLocation!!.bearingTo(location.toAndroidLocation()) - userHeading!!
+                            )
+                            Column(horizontalAlignment = Alignment.CenterHorizontally) {
+                                Icon(
+                                    modifier = Modifier
+                                        .padding(top = 8.dp, bottom = 8.dp, start = 16.dp)
+                                        .rotate(directionArrowAngle),
+                                    imageVector = Icons.Rounded.ArrowUpward,
+                                    contentDescription = null
+                                )
+                                if (distance != null) {
+                                    Text(
+                                        text = distance.metersToString(context)
```

Comment:
There should be an option to choose between sane and imperial units somewhere.

Reply:
You are right, I forgot about users of insane units :eyes: